### PR TITLE
improved lock and mm api

### PR DESCRIPTION
**Pull request by nadvornik**
depends on https://github.com/os-autoinst/openQA/pull/449

### CODE REVIEW

File: lockapi.pm [hunk 1]
```diff
@@ -20,36 +20,66 @@ use strict;
...
 
-sub mutex_lock($) {
-    my ($name) = @_;
-    return _mutex_call('post', "mutex/$name/lock");
+sub mutex_lock($;$) {
+    my ($name, $where) = @_;
+    bmwqemu::diag("mutex lock '$name'");
+    while (1) {
+        my $param = {action => 'lock'};
+        $param->{where} = $where if $where;
```
Comment (aaannz) [inline lockapi.pm]: Shouldn't there be some default for `where`? Since for mutex_lock it is mandatory.

File: lockapi.pm [hunk 2]
```diff
@@ -20,36 +20,66 @@ use strict;
...
+        sleep(5);
+    }
+}
+
+sub mutex_try_lock($;$) {
+    my ($name, $where) = @_;
+    bmwqemu::diag("mutex try lock '$name'");
+    my $param = {action => 'lock'};
+    $param->{where} = $where if $where;
+    my $res = api_call('post', "mutex/$name", $param)->code;
```
Comment (aaannz) [inline lockapi.pm]: again as above

**Comment by aaannz** [comment]
the rest LGTM

File: lockapi.pm (hunk 1 above)
Comment (nadvornik) [inline lockapi.pm]: missing where argument means the default - search self and parents, it is handled on server side

**Comment by coolo** [comment]
did you guys make up your mind?

**Comment by aaannz** [comment]
I think this is ok, but we need to merge https://github.com/os-autoinst/openQA/pull/449 first

**Comment by coolo** [comment]
please rebase

**Comment by coolo** [comment]
merged #329

**Pull request closed without merging**